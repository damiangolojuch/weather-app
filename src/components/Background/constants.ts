export const backgrounds = [
  "pexels-andy-vu-3244513.jpg",
  "pexels-andy-vu-3484061.jpg",
  "pexels-asad-photo-maldives-1591373.jpg",
  "pexels-frans-van-heerden-624015.jpg",
  "pexels-freestocksorg-839462.jpg",
  "pexels-lumn-167699.jpg",
  "pexels-pixabay-235621.jpg",
  "pexels-roberto-nickson-2559941.jpg",
  "pexels-roberto-nickson-2775196.jpg",
  "pexels-stein-egil-liland-3408744.jpg",
];
