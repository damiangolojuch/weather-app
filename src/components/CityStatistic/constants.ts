export enum CityStatisticType {
  windSpeed = "windSpeed",
  humidity = "humidity",
  pressure = "pressure",
  temperature = "temperature",
}

export const CityStatistics = {
  [CityStatisticType.humidity]: {
    name: "Wilgotność",
    suffix: "%",
  },
  [CityStatisticType.windSpeed]: {
    name: "Prędkość wiatru",
    suffix: "m/s",
  },
  [CityStatisticType.pressure]: {
    name: "Ciśnienie",
    suffix: "hPa",
  },
  [CityStatisticType.temperature]: {
    name: "Temperature",
    suffix: "°C",
  },
};
