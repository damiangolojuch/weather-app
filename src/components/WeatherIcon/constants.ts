export const enum WeatherIcons {
  cloudy = "cloudy.svg",
  "cloudyDay1" = "cloudy-day-1.svg",
  "cloudyDay2" = "cloudy-day-2.svg",
  "cloudyDay3" = "cloudy-day-3.svg",
  "cloudyNight1" = "cloudy-night-1.svg",
  "cloudyNight2" = "cloudy-night-2.svg",
  "cloudyNight3" = "cloudy-night-3.svg",
  day = "day.svg",
  night = "night.svg",
  "rainy1" = "rainy-1.svg",
  "rainy2" = "rainy-2.svg",
  "rainy3" = "rainy-3.svg",
  "rainy4" = "rainy-4.svg",
  "rainy5" = "rainy-5.svg",
  "rainy6" = "rainy-6.svg",
  "rainy7" = "rainy-7.svg",
  "snowy1" = "snowy-1.svg",
  "snowy2" = "snowy-2.svg",
  "snowy3" = "snowy-3.svg",
  "snowy4" = "snowy-4.svg",
  "snowy5" = "snowy-5.svg",
  "snowy6" = "snowy-6.svg",
  thunder = "thunder.svg",
  weather = "weather.svg",
  "weatherSprite" = "weather-sprite.svg",
  weatherSagittarius = "weather_sagittarius.svg",
  weatherSunset = "weather_sunset.svg",
}
