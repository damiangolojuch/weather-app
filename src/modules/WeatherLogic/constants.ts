export interface ApiCityWeatherType {
  cisnienie: string;
  data_pomiaru: string;
  godzina_pomiaru: string;
  id_stacji: string;
  kierunek_wiatru: string;
  predkosc_wiatru: string;
  stacja: string;
  suma_opadu: string;
  temperatura: string;
  wilgotnosc_wzgledna: string;
}

export interface CityWeatherType {
  id: string;
  city: string;
  windSpeed: number;
  humidity: number;
  pressure: number;
  temperature: number;
  clouds: number;
}
