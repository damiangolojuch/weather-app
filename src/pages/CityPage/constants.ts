export const compareToCities = ["Warszawa", "Kraków", "Rzeszów", "Gdańsk"];
