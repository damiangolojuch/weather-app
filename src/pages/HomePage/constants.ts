export const defaultCities = ["Warszawa", "Kraków", "Rzeszów"];
