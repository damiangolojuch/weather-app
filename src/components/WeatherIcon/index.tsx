import { FC } from "react";
import * as S from "./styles";
import { WeatherIcons } from "./constants";

export interface WeatherIconProps {
  type: WeatherIcons;
  className?: string;
}

const WeatherIcon: FC<WeatherIconProps> = ({ type, className }) => {
  const iconSrc = `${
    process.env.PUBLIC_URL || ""
  }/assets/weather_icons/animated/${type}`;
  return <S.Icon className={className} src={iconSrc} />;
};

export default WeatherIcon;
