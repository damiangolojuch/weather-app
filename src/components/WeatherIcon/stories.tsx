import { ComponentMeta, ComponentStory } from "@storybook/react";
import WeatherIcon from ".";
import { WeatherIcons } from "~/components/WeatherIcon/constants";

export default {
  title: "WeatherIcon",
  component: WeatherIcon,
  args: {
    description: "Weather icon component",
  },
} as ComponentMeta<typeof WeatherIcon>;

const DefaultTemplate: ComponentStory<typeof WeatherIcon> = (args) => (
  <WeatherIcon {...args} />
);

export const Default = DefaultTemplate.bind({});

Default.args = {
  type: WeatherIcons.cloudyDay1,
};
