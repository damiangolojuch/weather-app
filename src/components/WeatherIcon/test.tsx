import { render } from "@testing-library/react";
import { WeatherIcons } from "./constants";
import WeatherIcon, { WeatherIconProps } from ".";

const setup = (customProps: Partial<WeatherIconProps> = {}) => {
  const defaultProps: WeatherIconProps = {
    type: WeatherIcons.cloudyDay2,
    ...customProps,
  };

  return render(<WeatherIcon {...defaultProps} />);
};

describe("<WeatherIcon />", () => {
  it(`should render img with proper src`, () => {
    const { container } = setup({ type: WeatherIcons.weatherSprite });

    expect(container).toMatchSnapshot();
  });
});
