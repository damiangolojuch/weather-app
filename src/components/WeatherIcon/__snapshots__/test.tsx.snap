// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherIcon /> should render img with proper src 1`] = `
<div>
  <img
    class="sc-bczRLJ"
    src="/assets/weather_icons/animated/weather-sprite.svg"
  />
</div>
`;
